tsdb/meta.go

```diff
@@ -780,56 +781,43 @@ func (m *Measurement) TagSets(dimensions []string, condition influxql.Expr) ([]*
 		for _, dim := range dimensions {
 			tags[dim] = s.Tags.GetString(dim)
 		}
-
 		// Convert the TagSet to a string, so it can be added to a map allowing TagSets to be handled
 		// as a set.
-		tagsAsKey := string(MarshalTags(tags))
-		tagSet, ok := tagSets[tagsAsKey]
+		tagsAsKey := MarshalTags(tags)
+		tagSet, ok := tagSets[string(tagsAsKey)]
 		if !ok {
 			// This TagSet is new, create a new entry for it.
-			tagSet = &influxql.TagSet{}
-			tagsForSet := make(map[string]string, len(tags))
-			for k, v := range tags {
-				tagsForSet[k] = v
+			tagSet = &influxql.TagSet{
+				Tags: tags,
+				Key:  tagsAsKey,
 			}
-			tagSet.Tags = tagsForSet
-			tagSet.Key = MarshalTags(tagsForSet)
 		}
-
 		// Associate the series and filter with the Tagset.
 		tagSet.AddFilter(m.seriesByID[id].Key, filter)
 
 		// Ensure it's back in the map.
-		tagSets[tagsAsKey] = tagSet
-	}
-
-	// Sort the series in each tag set.
-	for _, t := range tagSets {
-		sort.Sort(t)
```

Comment:
The sort is moved to here when necessary: https://github.com/influxdata/influxdb/pull/7385/files#diff-82e4e65bd72f00208a49c4797d466ecdR1164